packages/plugin-data/test/testUtil.ts

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import {
+  BaseConnection,
+  Batch,
+  BatchInfo,
+  Job,
+  QueryResult,
+  RecordResult,
+  SaveResult,
+  SObject,
+  UpsertResult,
+  UserInfo,
+} from 'jsforce';
+import Parser = require('fast-xml-parser');
+import { Connection } from '@salesforce/core';
+
+import EventEmitter = NodeJS.EventEmitter;
+
+// needs an external _listeners object since its not included in the type definition
+export const createBaseFakeEmitter = function (): EventEmitter {
+  return {
+    on(event: string, listener: Function): EventEmitter {
+      return this;
+    },
+    emit(event: string, ...args: any[]): boolean {
+      return true;
+    },
+    addListener(event: string, listener: Function): EventEmitter {
+      return this;
+    },
+    once(event: string, listener: Function): EventEmitter {
+      return this;
+    },
+    removeListener(event: string, listener: Function): EventEmitter {
+      return this;
+    },
+    removeAllListeners(event?: string): EventEmitter {
+      return this;
+    },
+    setMaxListeners(n: number): EventEmitter {
+      return this;
+    },
+    getMaxListeners(): number {
+      return 10;
+    },
+    listeners(event: string): Function[] {
+      return [];
+    },
+    listenerCount(event: string): number {
+      return 0;
+    },
+    prependListener(event: string | symbol, listener: Function) {
+      return this;
+    },
+    prependOnceListener(event: string | symbol, listener: Function) {
+      return this;
+    },
+    off(event: string, listener: Function): EventEmitter {
+      return this;
+    },
+    rawListeners(event: string | symbol) {
+      return [];
+    },
+    eventNames(): Array<string | symbol> {
+      return [];
+    },
+  };
+};
+
+export const createBaseFakeConnection = function (): BaseConnection {
```

Comment:
do we really need all of these methods stubbed? same with `Emitter` above. and `Connection` below.

Reply:
We should be able to switch this to `stubInterface` or `stubObject` from `@salesforce/ts-sinon`.

Reply:
I will look into this, thanks.